share/p2p/peers/manager.go

```diff
@@ -348,11 +365,14 @@ func (m *Manager) cleanUp() []peer.ID {
 	addToBlackList := make(map[peer.ID]struct{})
 	for h, p := range m.pools {
 		if time.Since(p.createdAt) > m.poolValidationTimeout && !p.isValidatedDataHash.Load() {
+			delete(m.pools, h)
+			if p.headerHeight < m.firstHeader {
+				continue
+			}
```

Comment:
Yes. This code handles the corner case when hash is received from shrex.Sub before any headers from headerSub. Peer manager will create pool for such hash and if it will not get validated, we will just remove it. Note that after first header is received from headerSub such pool will no longer be created, messages will be ignored on Height check inside validator

Reply:
There are 2 test cases added to target different order of Hashes and headerSub.